# HG changeset patch
# User Vsevolod Volkov <vvv@mutt.org.ua>
# Date 1454260630 0
#      Sun Jan 31 17:17:10 2016 +0000
# Branch gentoo-1.7
# Node ID d075b5c2a28c4e6bf3c97300c3799990467fd158
# Parent  9fe3a802a546726352432446319a33ddbd7f7b72
[PATCH] feature: initials
From 556389fa3c9f7e7282863ae7b8a8b5a8588c86c7 Mon Sep 17 00:00:00 2001
Expando for author's initials
---
 hdrline.c | 23 +++++++++++++++++++++++
 1 file changed, 23 insertions(+)

diff -r 9fe3a802a546 -r d075b5c2a28c PATCHES
--- a/PATCHES	Tue Jan 26 23:00:09 2016 +0000
+++ b/PATCHES	Sun Jan 31 17:17:10 2016 +0000
@@ -1,3 +1,4 @@
+patch-initials-neomutt
 patch-ifdef-neomutt
 patch-browser-sort-additions-neomutt
 patch-bind-enter-return-separately-gentoo
diff -r 9fe3a802a546 -r d075b5c2a28c hdrline.c
--- a/hdrline.c	Tue Jan 26 23:00:09 2016 +0000
+++ b/hdrline.c	Sun Jan 31 17:17:10 2016 +0000
@@ -212,6 +212,7 @@ int mutt_user_is_recipient (HEADER *h)
  * %f = entire from line
  * %F = like %n, unless from self
  * %i = message-id
+ * %I = initials of author
  * %l = number of lines in the message
  * %L = like %F, except `lists' are displayed first
  * %m = number of messages in the mailbox
@@ -464,6 +465,28 @@ hdr_format_str (char *dest,
       mutt_format_s (dest, destlen, prefix, hdr->env->message_id ? hdr->env->message_id : "<no.id>");
       break;
 
+    case 'I':
+      {
+	int iflag = FALSE;
+	int j = 0;
+
+	for (i = 0; hdr->env->from && hdr->env->from->personal &&
+		    hdr->env->from->personal[i] && (j < (SHORT_STRING - 1)); i++) {
+	  if (isalpha ((int) hdr->env->from->personal[i])) {
+	    if (!iflag) {
+	      buf2[j++] = hdr->env->from->personal[i];
+	      iflag = TRUE;
+	    }
+	  } else {
+	    iflag = FALSE;
+	  }
+	}
+
+	buf2[j] = '\0';
+      }
+      mutt_format_s (dest, destlen, prefix, buf2);
+      break;
+
     case 'l':
       if (!optional)
       {
